pkg/service/backup/restore_worker_tables.go

```diff
@@ -563,6 +574,8 @@ func (w *tablesWorker) startDownload(ctx context.Context, host, dstDir, srcDir s
 		return 0, 0, errors.Wrap(err, "download batch to upload dir")
 	}
 
+	w.metrics.SetRestoreState(w.ClusterID, w.location, w.miwc.SnapshotTag, host, metrics.RestoreStateDownloading)
```

Comment:
I am not sure with the placement of setting `downloading` state. For example, in this fragment of `reactivateRunProgress`:
```go
// Nothing to do if rclone job is still running
if job, err := w.Client.RcloneJobProgress(ctx, pr.Host, pr.AgentJobID, w.Config.LongPollingTimeoutSeconds); err != nil {
	if scyllaclient.WorthWaitingForJob(job.Status) {
		return nil
	}
}
```
we could also be in `downloading` state even though we haven't reached `startDownload`. In my opinion we should preemptively set `downloading` state at the beginning of the host main for loop:
```go
for {
      pr, err := w.prepareRunProgress(ctx, run, target, h, dstDir, srcDir)
      // TODO: set downloading state here
```
Note that we have to do it after `prepareRunProgress` as host might wait there for ID's for its batch.

Reply:
It make sense actually. Thanks !

Reply:
Ok, I deleted previous comment confirming your statement.
But I bring it back, it makes sense to set the download here.